src/Compression/CompressedReadBufferBase.cpp

```diff
@@ -187,6 +187,79 @@ size_t CompressedReadBufferBase::readCompressedData(size_t & size_decompressed,
     return size_compressed_without_checksum + sizeof(Checksum);
 }
 
+size_t CompressedReadBufferBase::readCompressedDataBlockHold(size_t & size_decompressed, size_t & size_compressed_without_checksum)
```

Comment:
Just skimmed this method but it appears mostly copypasta of readCompressedData(). Could we unify them somehow?